docs: installation

naftis opened the pull request:
Initial installation documentation. I added it here for starters, we will probably move these to Gitbook.

naftis committed:
docs: installation

naftis committed:
chore: use the version tag

tahmidrahman-dsi reviewed (approved):
Looks good. 👍🏼

tahmidrahman-dsi commented:
Should we put id reader realated form configuration changes documented in this repo under `/docs`? @naftis

naftis committed:
chore: remove todo comments from would they throw

I dont think they would throw as the verifyNid API is called on-demand and the file is opened on demand in MOSIPAuthenticator

naftis committed:
Update installation.md

tahmidrahman-dsi commented:
> @tahmidrahman-dsi
> 
> Some non-obvious documentation things:
> 
> 1. External validation workqueue needs to be enabled from country-config default settings
> 
> I'll continue amending if more comes into mind..

Please feel free to have a look @naftis https://app.gitbook.com/o/zub8C4BetmW3a9Bj4Cd4/s/esn2q91OsFQf7ZqR8thb/~/revisions/UpoFPVYrQo17nWTN7grd/setup/3.-installation/3.2-set-up-your-own-country-configuration/3.2.5-set-up-application-settings#id-1.-prepare-global-functional-settings 
in the external validation workqueue section

The PR https://github.com/opencrvs/documentation/pull/21
cc: @euanmillar

naftis commented:
1. Install pip install jwcrypto (assuming you have Python installed)

2. Generate a private key PEM
openssl genpkey -algorithm RSA -pkeyopt rsa_keygen_bits:2048 -out private_key.pem
You might want to save this to a secure trusted place

**For OpenCRVS**

3. Generate a **Base64 private JWK** from the PEM. This always produces the same result from the same private key, no need to save this.
```bash
python3 -c "from jwcrypto import jwk; k = jwk.JWK.from_pem(open('private_key.pem','rb').read()); k.use = 'sig'; k.alg = 'RS256'; print(k.export(private_key=True))" | base64
```

**For MOSIP**

4. Generate **a public JWK** from the PEM. Also always produces same result.
```bash
python3 -c "from jwcrypto import jwk; k = jwk.JWK.from_pem(open('private_key.pem','rb').read()); k.use = 'sig'; k.alg = 'RS256'; print(k.export(private_key=False))"
```

naftis commented:
https://github.com/opencrvs/mosip/blob/2db6435d1af3d24d2536f9ece6e36ac4845fb613/packages/mosip-api/src/mosip-api.ts#L97
- [ ] Remember to documentate all REQUIRED values. The audit would for example throw here if not added.

naftis commented:
Closing as stale
